### ISSUE TITLE
Training Time

### ISSUE THREAD

**Issue by arunabh1904** [JindongJiang/RedNet#4]
Hi. So, in the paper you mention that you train on a 1080 with a batch size of 5. Can you tell me how long it took to train the network for Sun RGBD? I have a similar dataset and I am training it on 4*Tesla VX100's with a batch size of 20 and it takes 3-4 days for the network to train. Also, I am using a train-eval split of 80-20 and while my pixelwise accuracy for the training increases to about 95% my evaluation accuracy stays stagnant at about 50% Attached is a clipping from my tensorboard. 
![tensorboard](https://user-images.githubusercontent.com/38589274/48006015-dc6a8800-e0e2-11e8-9b23-da3aa1d33fe0.png)
If you have some insight as to why this is happening, it would be really helpful.

**Comment by JindongJiang** [comment]
Hi,
Actually, 3-4 days of training is reasonable, in our implementation, it took approximately 4 days to train the network. 
For the low acc in testing data, it is hard to know the exact reason without investigating your dataset, but there's something you can try. First, you can check if classes your dataset are unbalanced, you might want to use weighted loss for different classes. Also, if the distribution of visual feature for different classes is highly intersected, e.g., a class monitor or computer and a class television, you might want to merge the classes.

**Comment by arunabh1904** [comment]
The training and test dataset is super similar. My classes are imbalanced, but I am using the median frequency setting to weigh the cross entropy loss. Shouldnt that better it? All my classes are mutually exclusive. I dont think the visual features are overlapping. What do you mean when you say weighted loss fr different classes?

**Comment by arunabh1904** [comment]
Also, If while training on a subset of data one of the classes is missing, what kind of effect will that have on the network? Will it not be able to converge at all? Additionally, recently I have started training on a different dataset and the loss for training and validation seems to stay stagnant for the first 200 epochs. Is that something you have come across earlier? If not, do you have some insight to why that could be happening?

**Comment by JindongJiang** [comment]
Hi,

Sorry for my slow response because of several midterms. The "weighted loss fr different classes" is just a typo (fr = "FOR"), I actually mean the "weighted loss" which you have already implemented.

Since the deep-learning method nowadays is basically doing statistical inference, missing a class in your data may imply that the P(missing class|Image) is comparatively low, which might result in a low performance on that class on the test set, but this should not infect its property on convergence. 

The training loss staying stagnant actually happens a lot during the training of our network, you should try to keep training it longer to make sure it indeed stop decreasing, sometimes its helpful to increase the epochs number for learning rate decay.

Good luck.

**Comment by arunabh1904** [comment]
Thank you. I got it working. It was a bug in the code that I wrote. Good luck for your midterms.
